src/components.rs

```diff
@@ -142,18 +142,18 @@ pub trait Component {
 
     /// Set the [`DTSTART`](https://datatracker.ietf.org/doc/html/rfc5545#section-3.8.2.4) [`Property`]
     ///
-    /// See [`CalendarDateTime`] for info how are different [`chrono`] types converted automatically.
-    fn starts<T: Into<CalendarDateTime>>(&mut self, dt: T) -> &mut Self {
+    /// See [`DatePerhapsTime`] for info how are different [`chrono`] types converted automatically.
+    fn starts<T: Into<DatePerhapsTime>>(&mut self, dt: T) -> &mut Self {
```

Comment:
I'm happy to change the name if you can think of a better one, but it does need to be a different type than 'CalendarDateTime`, because some fields are required to be a DATE-TIME (and so I've used `CalenderDateTime`) whereas some can be either a DATE-TIME or just a DATE (which is what `DatePerhapsTime` is for).